src/pyodide/internal/workers.py

```diff
@@ -0,0 +1,76 @@
+# This module defines a Workers API for Python. It is similar to the API provided by
+# JS Workers, but with changes and additions to be more idiomatic to the Python
+# programming language.
+from http import HTTPMethod, HTTPStatus
+from typing import TypedDict, Unpack
+
+import js
+
+import pyodide.http
+from pyodide.http import pyfetch
+
+JSBody = (
+    "js.Blob | js.ArrayBuffer | js.TypedArray | js.DataView | js.FormData |"
+    "js.ReadableStream | js.URLSearchParams"
+)
+Body = "str | JSBody"
+Headers = dict[str, str] | list[tuple[str, str]]
+
+
+class FetchKwargs(TypedDict, total=False):
+    headers: Headers | None
+    body: "Body | None"
+    method: HTTPMethod = HTTPMethod.GET
+
+
+# TODO: This is just here to make the `body` field available as many Workers
+# examples which rewrite responses make use of it. It's something we should
+# likely upstream to Pyodide.
+class FetchResponse(pyodide.http.FetchResponse):
+    @property
+    def body(self) -> Body:
+        """
+        Returns the body from the JavaScript Response instance.
+        """
+        return self.js_response.body
+
+
+async def fetch(
+    resource: str,
+    **other_options: Unpack[FetchKwargs],
+) -> FetchResponse:
+    if "method" in other_options and isinstance(other_options["method"], HTTPMethod):
+        other_options["method"] = other_options["method"].value
+    resp = await pyfetch(resource, **other_options)
+    return FetchResponse(resp.url, resp.js_response)
+
+
+def Response(
+    body: Body,
+    status: HTTPStatus | int = HTTPStatus.OK,
+    statusText="",
+    headers: Headers = None,
+) -> js.Response:
```

Comment:
If the name is `Capitalized` then it should probably be a class, for instance with support for `isinstance`. If you want the result to be actually a `js.Response` you could add a `__instancecheck__` and a `__new__`. Or with the change to `python-entrypoint-helper` you could make a Python class with a `js_response` field. That may be the most reasonable approach. Maybe we should think about upstreamng changes to make it convenient to construct a `pyodide.ffi.FetchResponse` directly.